sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -79,6 +79,7 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
 }
 
 EventImplPtr Scheduler::addCopyBack(Requirement *Req) {
+  std::lock_guard<std::mutex> lock(MGraphLock);
```

Comment:
For a modern C++ mood:
```suggestion
  std::lock_guard<std::mutex> lock { MGraphLock };
```

Otherwise, too bad we are not using C++17:
```
  std::lock_guard lock { MGraphLock };
```

Reply:
@keryell other methods use older style. I’d prefer to keep it consistent.

> Otherwise, too bad we are not using C++17:

Customers, who can’t switch to a newer compiler, would disagree with you.

Reply:
@alexbatashev this is a sad story... I hope they can apply to another company where they can use modern C++ ;-)